numpy/core/src/multiarray/ctors.c

```diff
@@ -1766,7 +1766,8 @@ PyArray_FromAny(PyObject *op, PyArray_Descr *newtype, int min_depth,
     }
 
     if (cache == NULL && newtype != NULL &&
-            PyDataType_ISSIGNED(newtype) && PyArray_IsScalar(op, Generic)) {
+            PyDataType_ISSIGNED(dtype) &&
```

Comment:
Not sure why this never warned... it seems pretty harmless, but while `newtype != NULL` is fine, checking its content is in principle not.